static/js/stream_list.js

```diff
@@ -154,7 +154,11 @@ export function build_stream_list(force_rerender) {
         add_sidebar_li(stream_id);
     }
 
-    parent.append(elems);
+    if (elems.length > 0) {
+        parent.append(elems);
+    } else {
+        parent.append('<div class="stream-not-found"> No search results </div>');
```

Comment:
Strings should be tagged for translation, and ideally use a handlebars template; we don't like raw HTML in JavaScript like this.

Most of our similar components use a `data-empty` for this sort of string; can you try doing that?

Reply:
Yeah sure